warehouse/csp.py

```diff
@@ -32,13 +31,15 @@ def content_security_policy_tween(request):
 
 
 def csp_factory(_, request):
-    return copy.deepcopy(request.registry.settings.get("csp", {}))
+    return collections.defaultdict(
+        list, request.registry.settings.get("csp", {})
```

Comment:
I think you still want this to be a deepcopy else modifying the lists involved will persist I think.